### PR TITLE
refactor(rollout): centralize common group filters

### PR DESCRIPTION
**Pull request by guapisolo**
> **Depends on:** #3095
> Stacked on the parent PR's branch. Review / merge the parent first.

## Summary
Centralize rollout filters with explicit names and deprecated legacy entrypoints.

## Motivation
Stock filters and staleness calculation were spread across filter modules and `DefaultDataBuffer`. Centralizing their implementations and naming `FilterOutput` functions `apply_xxx_filter` makes ownership and call sites clearer while preserving existing CLI/configuration paths and runtime behavior.

## Before / After
- **Before / After:** `common_filters.py` owns aborted validation, reward-variance filtering, and staleness calculation; buffer actions remain in `DefaultDataBuffer`.
- **What moved where:** The stock filters become `apply_aborted_filter` and `apply_reward_nonzero_std_filter`; `group_staleness` remains a numeric helper.
- **Compatibility:** Both old names in `dynamic_sampling_filters.py` remain thin wrappers with their original signatures, returning the canonical result directly.
- **Deprecation:** `typing_extensions.deprecated(..., category=None)` marks only the legacy wrappers without runtime warnings; `typing_extensions>=4.5` is declared directly.
- **Buffer ownership:** Metrics, retry/drop handling, locking, and queue mutation retain their existing order; pre-put validation remains `aborted -> custom`.

## Behavior Preservation
- `test_filters.py` checks old dotted-path loading, structured keep/drop results, argument forwarding, unchanged exceptions, and deprecation metadata; canonical functions remain undecorated.
- Direct buffer probes use both old dotted paths to verify aborted recycling, reward-variance rejection, and valid nested-group enqueue/get; wrapper signatures match the original API in `58bad6e9`.

## Verification
- On `91d2ba61fb1262f889723d3284642a103c986882`, `python -m pytest --confcutdir=tests/fast/rollout tests/fast/rollout/test_filters.py -o addopts= -q`: 10 passed; repository-wide integration fixtures were not loaded.
- Direct buffer compatibility probes passed with deprecation warnings treated as errors.
- Ruff `0.14.7`, isort `5.13.2`, Black `24.3.0`, compilation, and `git diff --check` passed for the changed Python files.
- Full collector integration and GPU CI were not rerun locally.

## Review Focus
- `dynamic_sampling_filters.py`: legacy paths remain callable with unchanged results; only old wrappers carry deprecation metadata.
- `DefaultDataBuffer`: aborted/stale handling and side-effect ordering remain unchanged.
- `common_filters.py`: shared implementation ownership; missing-reward rejection belongs to the dependent PR.

### CODE REVIEW

File: miles/rollout/filter_hub/common_filters.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import torch
+
+from miles.rollout.filter_hub.base_types import FilterOutput, iter_samples
+from miles.utils.types import Sample
+
+Group = list[Sample | list[Sample]]
+
+
+def check_no_aborted(args: Namespace, samples: Group, **kwargs) -> FilterOutput:
```
Comment (yueming-yuan) [inline miles/rollout/filter_hub/common_filters.py]: nit: maybe add a "_filter" postfix to all three filters to make the code easier to read in other files?

**Review by yueming-yuan — APPROVED** [review]

File: miles/rollout/filter_hub/common_filters.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+def check_reward_nonzero_std(args, samples: list[Sample | list[Sample]], **kwargs):
+    rewards = [sample.get_reward_value(args) for sample in iter_samples(samples)]
+    keep = torch.tensor(rewards, dtype=torch.float64).std() > 1e-8
+    return FilterOutput(
+        keep=keep,
+        reason=None if keep else f"zero_std_{round(rewards[0], 1)}",
+    )
+
+
+def group_staleness(group: Group, current_version: int | None) -> int | None:
```
Comment (yueming-yuan) [inline miles/rollout/filter_hub/common_filters.py:27]: nit: maybe add a "_filter" postfix to all three filters to make it easier to understand what the function is in other files? (or considering the 3rd PR's change, call all functions in this shape, returning FilterOutput, `apply_xxx_filter`, wdyt)
